### ISSUE TITLE
"disk memory leak" when computing DF-RHF/RKS gradient

### ISSUE THREAD

**Issue by awhite862** [pyscf/pyscf#2388]
There is a temporary file created during the computation of density fitted rhf/rks gradients,
https://github.com/pyscf/pyscf/blob/44b2a20a9ba7545861c115cae98ee66f31e13340/pyscf/df/grad/rhf.py#L440

that is not necessarily cleaned up until the program exits. This can be a problem in geometry optimizations when the disk usage will increase with the number of times the gradient is computed.

One possible fix is to add a destructor to the `get_rhok_class` class defined in the same function:
```
+        def __del__(self):
+            self.f_rhok.close()
```

**Comment by sunqm** [comment]
Do you observe this on main branch or the 2.6.2 pip version? This issue should have been fixed in #2320 .

**Comment by awhite862** [comment]
I observed it in the 2.6.2 version, but it has indeed been fixed by #2320 , thanks!
